libraries/botbuilder-expression/src/buildInFunction.ts

```diff
@@ -970,6 +1137,76 @@ export class BuiltInFunctions {
                     BuiltInFunctions.VerifyContainer),
                 ReturnType.Number,
                 BuiltInFunctions.ValidateUnary),
+            new ExpressionEvaluator(
+                ExpressionType.Range,
+                BuiltInFunctions.ApplyWithError(
+                    (args: ReadonlyArray<any>) => {
+                        let error: string;
+                        if (args[1] < 1) {
+                            error = 'Second paramter must be more than zero';
+                        }
+
+                        let num: number;
+                        let result: number[] = [];
+                        for (num = 0; num < args[1]; num++) {
+                            result = result.concat(Number(args[0]) + num);
+                        }
+
+                        return { value: result, error };
+                    },
+                    BuiltInFunctions.VerifyInteger
+                ),
+                ReturnType.Object,
+                BuiltInFunctions.ValidateBinaryNumber
+            ),
+            new ExpressionEvaluator(
+                ExpressionType.Union,
+                BuiltInFunctions.Apply(
+                    (args: ReadonlyArray<any>) => {
+                        let result: any[] = [];
+                        for (const arg of args) {
+                            result = result.concat(arg);
+                        }
+
+                        return Array.from(new Set(result));
+                    },
+                    BuiltInFunctions.VerifyList),
+                ReturnType.Object,
+                BuiltInFunctions.ValidateAtLeastOne
+            ),
+            new ExpressionEvaluator(
+                ExpressionType.Intersection,
+                BuiltInFunctions.Apply(
+                    (args: ReadonlyArray<any>) => {
+                        let result: any[] = args[0];
+                        for (const arg of args) {
+                            result = result.filter((e: any) => arg.indexOf(e) > -1);
+                        }
+
+                        return Array.from(new Set(result));
+                    },
+                    BuiltInFunctions.VerifyList),
+                ReturnType.Object,
+                BuiltInFunctions.ValidateAtLeastOne
+            ),
+            new ExpressionEvaluator(
+                ExpressionType.Skip,
+                BuiltInFunctions.Skip,
+                ReturnType.Object,
+                (expression: Expression): void => BuiltInFunctions.ValidateOrder(expression, [], ReturnType.Object, ReturnType.Number),
+            ),
+            new ExpressionEvaluator(
+                ExpressionType.Take,
+                BuiltInFunctions.Take,
+                ReturnType.Object,
+                (expression: Expression): void => BuiltInFunctions.ValidateArityAndAnyType(expression, 2, 2, ReturnType.String, ReturnType.Object, ReturnType.Number),
```

Comment:
maybe `ValidateOrder` is more reasonable?

Reply:
yes. fixed